Made the labels in edit relationships tab unique

alexandrevryghem opened the pull request:
## Description
Currently, the relationship type labels are not unique and don't work well when the relationship type is not in the form `is{Entity1}Of{Entity2}`. This PR ensures they are unique by including the two entity types and the relationship type in the label.

## Instructions for Reviewers
List of changes in this PR:
* Changed the way the `getRelationshipMessageKey` works by always forming the labels in the form: `relationships.${currentEntityType}.${relationshipType}.${relatedEntityType}` (both entity types are required in order to support the use case from #1387)
* I mapped all the i18n values from the previous keys to the new keys to avoid breaking the relationship type labels in other languages
* I also changed the duplicate `Publication` label on the OrgUnit's edit relationship tab to: `Organisation Publications` & `Authored Publications`

**Guidance for how to test or review this PR:**
- Verify that all default keys are correctly translated (I only created translations for the ones defined in `dspace/config/entities/relationship-types.xml`)

## Checklist
- [x] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [x] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR **passes [ESLint](https://eslint.org/)** validation using `yarn lint`
- [x] My PR **doesn't introduce circular dependencies** (verified via `yarn check-circ-deps`)
- [x] My PR **includes [TypeDoc](https://typedoc.org/) comments** for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR **passes all specs/tests and includes new/updated specs or tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] My PR **aligns with [Accessibility guidelines](https://wiki.lyrasis.org/display/DSDOC8x/Accessibility)** if it makes changes to the user interface.
- [x] My PR **uses i18n (internationalization) keys** instead of hardcoded English text, to allow for translations.
- [x] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

tdonohue reviewed (commented):
@alexandrevryghem : We gave this a quick group review in our Developer Meeting today.  Overall, the logic here makes sense & I can understand how you'd want these i18n labels to be different for different pages/components.

The big challenge is that this touches a large number of i18n `json5` files, so it's going to hit frequent merge conflicts.  I think we'll need to find way to get this merge conflict free & test/merge rapidly for 9.0.  I'd hesitate to backport this PR because it changes so many i18n labels, but it seems good for 9.0.

Overall, there were no objections to this approach & it makes sense.  We just need to try to merge it quickly once this has merge conflicts resolved, as it will hit frequent merge conflicts.

alexandrevryghem commented:
@tdonohue: Sorry for the delay, I lost track of this issue, but all merge conflicts should be solved right now

tdonohue reviewed (changes requested):
@alexandrevryghem : I gave this a quick test today, and while the basics work, I'm now seeing some **new** relationships appearing when editing Entities.

For instance, this now appears at the bottom of the Relationships tab when editing a Person Entity:
![personEntity](https://github.com/user-attachments/assets/a9e99cf6-06ad-4b24-95cf-801700569f5a)

And, similarly, at the bottom of the Relationships tab when editing a Publication Entity:
![publicationEntity](https://github.com/user-attachments/assets/13cb5c34-a48e-4ba4-a5a1-8c012cbf5d82)

Is this expected behavior of this PR?  If so, it seems like we need to translate these new keys that have appeared under some Entities.

alexandrevryghem commented:
@tdonohue: Thnx for the review! This should be fixed now, this was because I tested this with a DB that only contained the default relationship types. I added the relationship types from `openaire4-relationships.xml` & `rioxx3-relationships.xml` locally and added the missing translations. After adding all the relationships from those files to the DB, I also discovered that there was another relationship label missing for one of the OrgUnit - Project relationships: `isFundingAgencyOfProject`/`isProjectOfFundingAgency`

tdonohue commented on src/assets/i18n/en.json5:
```diff
@@ -4307,18 +4307,26 @@
 
   "relationships.Person.isOrgUnitOfPerson.OrgUnit": "Organizational Units",
 
+  "relationships.Person.isPublicationOfContributor.Publication": "Publications",
```
When editing a Person Entity, you'll see *two* relationships that say "Publications".  This one, and the one above for `isPublicationOfAuthor`.  

Therefore, I'd recommend we give this a better name.  Maybe "Publications (contributor to)"?  We need to somehow differentiate these two different relationships in the user interface.

tdonohue commented on src/assets/i18n/en.json5:
```diff
@@ -4327,6 +4335,10 @@
 
   "relationships.OrgUnit.isPublicationOfAuthor.Publication": "Authored Publications",
 
+  "relationships.OrgUnit.isPublicationOfContributor.Publication": "Publications",
```
Same as above, this should likely be something like "Publications (contributor to)".

tdonohue commented on src/assets/i18n/en.json5:
```diff
@@ -4327,6 +4335,10 @@
 
+  "relationships.OrgUnit.isPublicationOfContributor.Publication": "Publications",
+
+  "relationships.OrgUnit.isProjectOfFundingAgency.Project": "Research Projects",
```
This also needs differentiation from `isProjectOfOrgUnit`.  Maybe it should be something like "Researcher Projects (funder of)"?

tdonohue reviewed (commented):
@alexandrevryghem : Thanks for the updates!  This mostly looks good, but I have a few **very minor** suggestions to change the text of some of the new i18n keys you've added.  They simply need a better description in some cases.

Beyond that, I'm +1 this PR.  I think it looks good and I've not noticed any issues.

alexandrevryghem commented:
@tdonohue: Thnx! I amended those 3 translation suggestions in the `en.json5` file

tdonohue reviewed (approved):
👍 Thanks @alexandrevryghem !  This looks good to me now.  All prior feedback has been addressed, and I'm not seeing any obvious bugs / side effects.
